### PR TITLE
Feature/wall jump

### PR DESCRIPTION
**Pull request by vincentpierre**
Made the Wall jump work
Renamed some of the shared assets

### CODE REVIEW

File: unity-environment/Assets/ML-Agents/Examples/WallJump/Scripts/WallJumpAcademy.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Examples/WallJump/Scripts/WallJumpAcademy.cs]: save tabs as spaces

File: unity-environment/Assets/ML-Agents/Scripts/Agent.cs [hunk 2]
```diff
@@ -841,6 +841,7 @@ void AgentStep()
     {
         if (terminate)
         {
+            stepCount = 0;
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Scripts/Agent.cs]: call this change out in the PR description.

File: unity-environment/Assets/ML-Agents/Examples/WallJump/Scripts/WallJumpAgent.cs [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+        Vector3 moveToPos = targetPos - rb.worldCenterOfMass;  
+        Vector3 velocityTarget = moveToPos * targetVel * Time.fixedDeltaTime;
+        if (float.IsNaN(velocityTarget.x) == false)
+        {
+            rb.velocity = Vector3.MoveTowards(
+                rb.velocity, velocityTarget, maxVel);
+        }
+    }
+
+    public void RayPerception(float rayDistance,
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Examples/WallJump/Scripts/WallJumpAgent.cs]: seems like this could benefit from @awjuliani PR where he refactored the perception change ... I would incorporate and then send out another PR.

File: unity-environment/Assets/ML-Agents/Examples/WallJump/Scripts/WallJumpAgent.cs [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+        groundMaterial = groundRenderer.material;
+
+        spawnArea.SetActive(false);
+    }
+
+
+    // Begin the jump sequence
+    public void Jump()
+    {
+
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Examples/WallJump/Scripts/WallJumpAgent.cs]: remove random empty lines from this file

File: unity-environment/Assets/ML-Agents/Examples/WallJump/Scripts/WallJumpAgent.cs (hunk 3 above)
Comment (vincentpierre) [reply unity-environment/Assets/ML-Agents/Examples/WallJump/Scripts/WallJumpAgent.cs]: Not until merged into development-0.3
